### DISCUSSION TITLE
VERY SLOW model loading time up to 20 minutes?? Something happened?

### DISCUSSION THREAD

**Discussion by eroxconsulting** [lllyasviel/Fooocus#3585] [Q&A]
Hey guys, so yesterday I installed Fooocus on my Macbook M3 pro 2024 13in, which should be more than capable to use this with 0 issues and I tried to generate images and it takes forever, Im not sure why but I need help. does anyone has these problem? If so how can I fixed it on Mac

**Comment by Crocodilian12** [comment]
_Hey guys, so yesterday I installed Fooocus on my Macbook M3 pro 2024 13in, which should be more than capable to use this with 0 issues and I tried to generate images and it takes forever, Im not sure why but I need help. does anyone has these problem? If so how can I fixed it on Mac_
---------------------------

Nope, not "more than capable" -- not at _all_ capable. Its a wonder if you get it going at all.

Mac silicon is simply does not remotely compare to a big Nvidia GPU for Stable Diffusion . . . nothing to do with Fooocus specifically (Fooocus is just the front end, a UI like ComfyUI, A1111 and so on . . . they all are front ends that push the full prompt to applications like Pytorch and Tensorflow, which have been written with CUDA hooks for efficient implementation on an Nvidia GPU. With some work, you can -- sometimes -- get an OK experience on a high spec AMD GPU

. . . but not on a Mac.

As a Mac user, I run Fooocus on a cloud service (RunDiffusion). That's the only way for a Mac user to have an acceptable experience with any Stable Diffusion UI. There is a native Mac Diffusion algorithm "Diffusion Bee" -- but the quality doesn't begin to match Stable Diffusion.

**Comment by FeliciaFiction** [comment]
> Hey guys, so yesterday I installed Fooocus on my Macbook M3 pro 2024 13in, which should be more than capable to use this with 0 issues and I tried to generate images and it takes forever, Im not sure why but I need help. does anyone has these problem? If so how can I fixed it on Mac

What kind of GPU does it have? If it's not Cuda or Vulkan capable you're gonna have a bad time...

To give you an example:
Running on a Ryzen 7700X (8 core, 16 threads) with 64GB ram @4 to 5GHz, it takes about 52 to 55 seconds per iteration (as an indication, it takes about 25 to 28GB of the available RAM)
Running with Cuda enabled Nvidia GTX 1650 Ti (4GB Vram) / Ryzen 7 4800H @1.4GHz it takes around 5 minutes (~10sec per iteration)
Running on a RTX 4070 Super (12GB Vram), it takes about 8 seconds (3.5sec per iteration)
